kornia/augmentation/_2d/intensity/plasma.py

```diff
@@ -0,0 +1,177 @@
+from typing import Any, Dict, Optional, Tuple, cast
+
+from torch import Tensor
+
+from kornia.augmentation import random_generator as rg
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.contrib import diamond_square
+
+
+class RandomPlasmaBrightness(IntensityAugmentationBase2D):
+    r"""Adds brightness to the image based on a fractal map generated by the diamond square algorithm.
+
+    .. image:: _static/img/RandomPlasmaBrightness.png
+
+    This is based on the original paper: TorMentor: Deterministic dynamic-path, data augmentations with fractals.
+    See: :cite:`tormentor` for more details.
+
+    .. note::
+        This function internally uses :func:`kornia.contrib.diamond_square`.
+
+    Args:
+        roughness: value to scale during the recursion in the generation of the fractal map.
+        intensity: value that scales the intensity values of the generated maps.
+        same_on_batch: apply the same transformation across the batch.
+        p: probability of applying the transformation.
+        keepdim: whether to keep the output shape the same as input (True) or broadcast it
+                 to the batch form (False).
+
+    Examples:
+        >>> rng = torch.manual_seed(0)
+        >>> img = torch.ones(1, 1, 3, 4)
+        >>> RandomPlasmaBrightness(roughness=(0.1, 0.7), p=1.)(img)
+        tensor([[[[0.6415, 1.0000, 0.3142, 0.6836],
+                  [1.0000, 0.5593, 0.5556, 0.4566],
+                  [0.5809, 1.0000, 0.7005, 1.0000]]]])
+    """
+
+    def __init__(
+        self,
+        roughness: Tuple[float, float] = (0.1, 0.7),
+        intensity: Tuple[float, float] = (0.0, 1.0),
+        same_on_batch: bool = False,
+        p: float = 0.5,
+        keepdim: bool = False,
+        return_transform: Optional[bool] = None,
+    ) -> None:
+        super().__init__(
+            p=p, return_transform=return_transform, same_on_batch=same_on_batch, p_batch=1.0, keepdim=keepdim
+        )
+        self._param_generator = cast(
+            rg.PlainUniformGenerator,
+            rg.PlainUniformGenerator((roughness, "roughness", None, None), (intensity, "intensity", None, None)),
```

Comment:
Why it is none here? It should be the range.

Reply:
the range is not specified by by passing the tuple ?

Reply:
Ah yes. I mistaken.

Reply:
Will be useful when the distribution outputs with inf, say Gaussian.